### DISCUSSION TITLE
Using multiple arguments?

### DISCUSSION THREAD

**Discussion by rsheppard-dev** [next-safe-action/next-safe-action#182] [Q&A]
I have two arguments on a function. How do I get them both into the action without combining them? I tried below but it did not work. I couldn't see this mentioned in docs as all actions seem to just have one argument.

Also what if you have no arguments?

```
export const myFunctionAction = action([myFunctionSchema, z.boolean()], myFunction);

async function myFunction(argument1: myFunctionSchema, argument2: boolean) {
	...function code...
}
```

Appreciate any help.

**Comment by TheEdoRan** [comment]
> I have two arguments on a function. How do I get them both into the action without combining them? I tried below but it did not work. I couldn't see this mentioned in docs as all actions seem to just have one argument.

Yes, just one main schema/argument is supported at this time. Multiple [bind args](https://nextjs.org/docs/app/building-your-application/data-fetching/server-actions-and-mutations#passing-additional-arguments) are supported.

> Also what if you have no arguments?

You can just omit the schema by not using the `schema` method when you're defining the action. The resulting client input will be typed `void`.
